docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1108 @@
+{
```

Comment:
<p><strong>Eugene:</strong></p><blockquote>Why is it important for the reader to know that protocols are pervasive in S4TF? Will this knowledge (in comparison with the knowledge that you can extend something called Layer whatever this is) help them develop better models? Or the goal is to make the reader feel good about S4TF because it uses advanced engineering techniques such as protocol-oriented programming?</blockquote><p><strong>Bart:</strong></p><blockquote>Ya I wasn't sure how I feel about this section when I wrote it. I mentioned in the original PR&nbsp;<a href="https://www.google.com/url?q=https://github.com/tensorflow/swift/pull/187&amp;sa=D&amp;ust=1558631981042000&amp;usg=AFQjCNGRTR6U8rx-kpbj-W48PsPodk6-GA" target="_blank">https://github.com/tensorflow/swift/pull/187</a>&nbsp;that I'm on the fence as to whether this section is necessary.</blockquote><blockquote><br></blockquote><blockquote>I thought it would have been nice to add something here to tie why we are talking about protocols in the Swift for TensorFlow repo/project, but similarly when we introduce Swift in the "Intro to Swift" Notebook, we don't tie it back to TensorFlow in any way.</blockquote><blockquote><br></blockquote><blockquote>So let me know what you think about completely removing this section as I don't see it adding much value.</blockquote>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Reply:
Would be open to any other ideas before I remove this section since it currently isn't that helpful